pkg/specgen/generate/container_create.go

```diff
@@ -136,13 +138,44 @@ func MakeContainer(ctx context.Context, rt *libpod.Runtime, s *specgen.SpecGener
 		options = append(options, libpod.WithNetworkAliases(s.Aliases))
 	}
 
+	if len(s.Devices) > 0 {
+		opts = extractCDIDevices(s)
+		options = append(options, opts...)
+	}
+
 	runtimeSpec, err := SpecGenToOCI(ctx, s, rt, rtc, newImage, finalMounts, pod, command)
 	if err != nil {
 		return nil, err
 	}
 	return rt.NewContainer(ctx, runtimeSpec, options...)
 }
 
+func extractCDIDevices(s *specgen.SpecGenerator) []libpod.CtrCreateOption {
+	devs := make([]spec.LinuxDevice, 0, len(s.Devices))
+	var cdiDevs []string
+	var options []libpod.CtrCreateOption
+
+	for _, device := range s.Devices {
+		isCDIDevice, err := cdi.HasDevice(device.Path)
+		if err != nil {
+			// do nothing
```

Comment:
should we at least log the error (warning or debug level)?

Reply:
This "error" will occur whenever the `--device` passed is not a CDI device.
Agreed it's worth logging somewhere, added.